kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java

```diff
@@ -171,39 +171,9 @@ private KaldbServiceGrpc.KaldbServiceFutureStub getKaldbServiceGrpcClient(String
   }
 
   @VisibleForTesting
-  public static Collection<String> getSearchNodesToQuery(
-      SnapshotMetadataStore snapshotMetadataStore,
-      SearchMetadataStore searchMetadataStore,
-      DatasetMetadataStore datasetMetadataStore,
-      long queryStartTimeEpochMs,
-      long queryEndTimeEpochMs,
-      String dataset) {
-    ScopedSpan findPartitionsToQuerySpan =
-        Tracing.currentTracer()
-            .startScopedSpan("KaldbDistributedQueryService.findPartitionsToQuery");
-
-    List<DatasetPartitionMetadata> partitions =
-        DatasetPartitionMetadata.findPartitionsToQuery(
-            datasetMetadataStore, queryStartTimeEpochMs, queryEndTimeEpochMs, dataset);
-    findPartitionsToQuerySpan.finish();
-
-    // step 1 - find all snapshots that match time window and partition
-    ScopedSpan snapshotsToSearchSpan =
-        Tracing.currentTracer().startScopedSpan("KaldbDistributedQueryService.snapshotsToSearch");
-    Set<String> snapshotsToSearch = new HashSet<>();
-    for (SnapshotMetadata snapshotMetadata : snapshotMetadataStore.getCached()) {
-      if (containsDataInTimeRange(
-              snapshotMetadata.startTimeEpochMs,
-              snapshotMetadata.endTimeEpochMs,
-              queryStartTimeEpochMs,
-              queryEndTimeEpochMs)
-          && isSnapshotInPartition(snapshotMetadata, partitions)) {
-        snapshotsToSearch.add(snapshotMetadata.name);
-      }
-    }
-    snapshotsToSearchSpan.finish();
-
-    // step 2 - iterate every search metadata whose snapshot needs to be searched.
+  protected static Set<String> getMatchingSearchMetadata(
+      SearchMetadataStore searchMetadataStore, Map<String, SnapshotMetadata> snapshotsToSearch) {
+    // iterate every search metadata whose snapshot needs to be searched.
     // if there are multiple search metadata nodes then pck the most on based on
     // pickSearchNodeToQuery
     ScopedSpan pickSearchNodeToQuerySpan =
```

Comment:
just spotted this while reviewing - I think this is a previous bug but can we call `pickSearchNodeToQuerySpan#finish` like we do to other spans when the calculation is done?